docs/architecture/adr-006-hermes-v0.2-usecases.md

```diff
@@ -0,0 +1,203 @@
+
+## Changelog
+* 16.02.2021: Proposed.
+
+## Context
+
+One major problem with planning for the evolution of Hermes, is that presently
+there is insufficient clarity regarding its requirements.
+It is not known who are the typical Hermes users (is it human operators or
+automated pipelines?), and what are their primary use-cases.
+
+This ADR proposes a few use-cases that seem interesting from the point
+of view of a general target base of users, and which will
+hopefully be a subset of the requirements of (any) future users.
+
+Three elements that provide further context for this discussion are:
+
+1. Hermes is still at an early stage of implementation, so these use-cases are
+   not set in stone.
+
+2. Some concrete use-cases are starting to emerge ([#628][628]), which we either
+   do not cover altogether, or cover poorly (e.g., because of inconsistent UX),
+   thus informing this proposal. 
+
+3. Hermes is one of _three_ relayer binaries that are being developed roughly in
+parallel. The other two are being developed in Go and Typescript, 
+respectively (see the [references](#references) section).
+In this context, it is plausible that Hermes will focus on performance,
+robustness, and richness of features on a longer term.
+
+## Decision
+
+The primary goal for the uses-cases we decided to cover is to prevent situations
+where users could get stuck. For example, the output of a command may be
+unclear, or there may be an error and thereby some CLI command
+finishes partially, or two relayers concurrently try to perform some
+operation(s) and interfere with each other, resulting in a chain state that is
+obscure to the user, and then the user could consequently be stuck.
+
+The first of the patterns below seeks to help "unblock" a user.
+The second pattern is a variation on the first; this permits more efficiency
+because it allows the reuse of previously-created objects in the
+creation of new objects on a chain (e.g., reuse a client in the creation of a
+connection).
+
+### Patterns
+
+We propose two basic patterns that Hermes should be able to fulfil.
+
+1. Simple invocations to perform basic actions.
+    - By _action_ here we mean doing the complete handshake for an object
+      (specifically _connection_ or _channel_) on two chains, or relaying
+      packets between two chains.
+    - The focus here is for the command to include retrying mechanisms 
+      (perform it _robustly_) and have the simplest interface.
+
+2. Allow reusing of pre-existing state for basic commands.
+    - The pre-existing state could be a client with some specific trust options,
+      for instance, and in this case Hermes would provide support for creating
+      a connection that uses this specific client.
+    - This pattern should also include a retrying mechanism.
+
+### Concrete Use-Cases
+
+Applying the above patterns to a few cases, we get the following concrete
+commands that Hermes v0.2.0 should fulfil.
+
+#### Create New Connection
```

Comment:
Could you please add the specifications for client create and update? I know we have them under `tx raw` but those are meant for testing and have extraneous and in the same time missing parameters.

Reply:
Done. Please have a look if the optional client parameters look good. Not sure if we need the height option for "client update" command.